main branch ("3.2") is failing pinned build

On ubuntu 22

```
1 warning generated.
In file included from /Workspaces/leap/unittests/state_history_tests.cpp:5:
In file included from /Workspaces/leap/libraries/state_history/include/eosio/state_history/log.hpp:8:
In file included from /Workspaces/leap/dependencies/boost_1_70_0/bin/include/boost/thread.hpp:13:
In file included from /Workspaces/leap/dependencies/boost_1_70_0/bin/include/boost/thread/thread.hpp:12:
In file included from /Workspaces/leap/dependencies/boost_1_70_0/bin/include/boost/thread/thread_only.hpp:17:
/Workspaces/leap/dependencies/boost_1_70_0/bin/include/boost/thread/pthread/thread_data.hpp:60:5: error: function-like macro '__sysconf' is not defined
#if PTHREAD_STACK_MIN > 0
    ^
/usr/include/x86_64-linux-gnu/bits/pthread_stack_min-dynamic.h:26:30: note: expanded from macro 'PTHREAD_STACK_MIN'
#   define PTHREAD_STACK_MIN __sysconf (__SC_THREAD_STACK_MIN_VALUE)
                             ^
1 error generated.
make[2]: *** [unittests/CMakeFiles/unit_test.dir/build.make:487: unittests/CMakeFiles/unit_test.dir/state_history_tests.cpp.o] Error 1
```

--- a/libraries/state_history/include/eosio/state_history/log.hpp
+++ b/libraries/state_history/include/eosio/state_history/log.hpp
@@ -5,7 +5,6 @@
 #include <stdint.h>
 
 #include <boost/asio.hpp>
-#include <boost/thread.hpp>
 
 #include <eosio/chain/block_header.hpp>
 #include <eosio/chain/exceptions.hpp>
@@ -156,11 +155,8 @@ class state_history_log {
    ~state_history_log() {
       // complete execution before possible vacuuming
       if (thr.joinable()) {
-         try  { 
-            work_guard.reset();
-            thr.join(); 
-         }
-         catch (const boost::thread_interrupted&) {/* suppressed */}
+         work_guard.reset();
+         thr.join();
       }     
 
       //nothing to do if log is empty or we aren't pruning

